src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -466,39 +444,77 @@ std::pair<vector<SparqlTripleSimple>, ParsedQuery::GraphPattern> Visitor::visit(
         AD_CORRECTNESS_CHECK(triple.p_.isVariable() || triple.p_.isIri());
         return SparqlTriple::fromSimple(triple);
       };
+  std::vector<Variable> visibleVariablesSoFar = std::move(visibleVariables_);
+  visibleVariables_.clear();
```

Comment:
I think this is only needed for subqueries, and updates can'tj be nested in something.
Or do you need this for the consecutive updates? These should probably be handled externally (we can talk abou this).

in thiese cases you could just add a check, the the visibleVariables are empty here.